buildenv/jenkins/JenkinsfileBase

```diff
@@ -1505,16 +1506,22 @@ def triggerRerunJob (rerunTestJobParams) {
 				if (rerunTestJobs) {
 					echo "Triggering rerun jobs in parallel ..."
 					def childJobs = parallel rerunTestJobs
-					archiveChildJobTap(childJobs)
+					def rerunResults = archiveChildJobTap(childJobs, originalStatus)
+					currentBuild.result = rerunResults.jobStatus
+				} else {
+					echo " Either all rerun succeed or there are aborted?"
+					//Here need to set currentBuild.result to be worst of non rerun parallel or rerun parallel 
 				}
 			}
 		}
 	}
 }
 
-def archiveChildJobTap(childJobs) {
+// update to return worst child job status, and failure test job parameters.
+def archiveChildJobTap(childJobs, originalStatus = 'SUCCESS') {
```

Comment:
I'm not entirely sure this implements quite https://github.com/adoptium/aqa-tests/issues/6000#issuecomment-2797842582
eg:
If "originalStatus" is say UNSTABLE,
we kick off rerun and say that is SUCCESS
then calls archiveChildJobTap(rerunChildJobs, UNSTABLE):
    currentStatus initialized to originalStatus which is UNSTABLE
    childRerunJob (SUCCESS) is not worse than currentStatus
    and currentStatus(UNSTABLE) is not worse than originalStatus(UNSTABLE)
so I think this ends up returning: jobStatus: currentStatus(UNSTABLE)

?

Reply:
Yes, currentStatus should be 'SUCCESS', it's been changed back and forth and ending up setting it incorrectly.  See my update.

Reply:
looks good now, thanks